blockchain/fastsync.go

```diff
@@ -213,38 +213,69 @@ func worker(cache *utxoCache, jobs <-chan []byte, results chan<- *result) {
 // downloadUtxoSet will attempt to connect to make an HTTP GET request to the
 // provided sources one at a time and download the UTXO set to the provided path.
 // If a proxy is provided it will use it for the HTTP connection.
-func downloadUtxoSet(sources []string, proxy *socks.Proxy, pth string) (string, error) {
-	var fileName string
-
-	grab.SetDefaultClientWithProxy(proxy)
-
-	for _, src := range sources {
-		log.Infof("Downloading UTXO set from %s", src)
-		err := retry.Do(
-			func() error {
-				resp, err := grab.Get(pth, src)
-				if err != nil {
-					return err
+func downloadUtxoSet(sources []string, proxy *socks.Proxy, destination string) (string, error) {
+	for _, source := range sources {
+		log.Infof("FastSync: Downloading UTXO set from %s to %s ...", source, destination)
+		retries := 3
+		for retries > 0 {
+			retries--
+
+			client := grab.NewClientWithProxy(proxy)
+			request, _ := grab.NewRequest(destination, source)
+
+			// Begin download
+			response := client.Do(request)
+
+			if response.HTTPResponse != nil {
+				fmt.Printf("FastSync: %v\n", response.HTTPResponse.Status)
+			}
+
+			displayDownloadProgress(response)
+
+			// Error checking
+			if responseError := response.Err(); responseError != nil {
+				log.Errorf("FastSync: Failed downloading UTXO set: %s", responseError.Error())
+				if strings.Contains(responseError.Error(), "connection refused") {
+					break
+				} else {
+					continue
 				}
+			}
 
-				fileName = resp.Filename
+			filename := response.Filename
+			log.Infof("\nFastSync: Download saved to %v", filename)
 
-				return nil
-			},
-			retry.Attempts(3),
-			retry.RetryIf(func(err error) bool {
-				return !strings.Contains(err.Error(), "connection refused")
-			}),
-		)
-
-		if err != nil {
-			log.Errorf("Failed downloading UTXO set: %s", err.Error())
-			continue
+			log.Info("FastSync: UTXO download complete. Verifying integrity...")
+			return filename, nil
 		}
-
-		log.Info("UTXO download complete. Verifying integrity...")
-		return fileName, nil
 	}
+	return "", AssertError("FastSync: All UTXO sources are unavailable")
+}
+
+// displayDownloadProgress will display progress text while a file is being downloaded
+// The text is displayed on the same line by utilising the '\r' character
+func displayDownloadProgress(resp *grab.Response) {
 
-	return "", AssertError("all UTXO sources are unavailable")
+	printer := message.NewPrinter(message.MatchLanguage("en"))
+	// Begin UI loop
+	timer := time.NewTicker(500 * time.Millisecond)
```

Comment:
NIT: This might be fine but I would probably do every second. =)

Reply:
Every second would be better.

Through usage we may want to change it to 5 seconds perhaps

Reply:
I would be fine with every 5 seconds. I think that is sufficient feedback for the user tbh.

Reply:
Updated to 5 seconds